ProjectLighthouse/Migrations/20230714212153_AddUserIdAndSlotIdToComment.cs

```diff
@@ -0,0 +1,46 @@
+﻿using LBPUnion.ProjectLighthouse.Database;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace ProjectLighthouse.Migrations
+{
+    [DbContext(typeof(DatabaseContext))]
+    [Migration("20230714212153_AddUserIdAndSlotIdToComment")]
+    public partial class AddUserIdAndSlotIdToComment : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
+                name: "TargetSlotId",
+                table: "Comments",
+                type: "int",
+                nullable: true);
+
+            migrationBuilder.AddColumn<int>(
+                name: "TargetUserId",
+                table: "Comments",
+                type: "int",
+                nullable: true);
+
+            // Set SlotId and UserId to null if there isn't an existing slot or user
+            migrationBuilder.Sql("UPDATE Comments AS c INNER JOIN Users AS u ON c.TargetId = u.UserId AND c.Type = 0 SET c.TargetUserId = u.UserId");
+            migrationBuilder.Sql("UPDATE Comments AS c INNER JOIN Slots AS s ON c.TargetId = s.SlotId AND c.Type = 1 SET c.TargetSlotId = s.SlotId");
+
+            // Delete rows that have null SlotIds or UserIds
+            migrationBuilder.Sql("DELETE FROM Comments WHERE TargetUserId IS NULL AND TargetSlotId IS NULL");
```

Comment:
Would this be beneficial to implement as an admin maintenance feature, sort of how you can delete broken photos? This can be applied to other areas as well.

Reply:
> This has the benefits of indexing, and cascading deletes so that when a user or level is deleted all of the comments are also deleted with it.

Unsure if above would defeat the purpose of this but might still be useful?

Reply:
By creating a foriegn key with a cascading delete, MySQL will automatically delete any comments that are on a user/level when it is deleted. Since there were no constraints before, some of the comments are on deleted levels or users and need to be manually deleted for the foreign key to be successfully created.